### PR TITLE
LibWeb+LibIDL+BindingsGenerator: DOMPoint Stuff and Things

### PR DESCRIPTION
**Pull request by AtkinsSJ**
This is a resurrection of some work I started in April. Since then, DOMPoint got added already, so the only new thing added to that is `fromPoint()`.

Otherwise, there are some improvements to BindingsGenerator.

### CODE REVIEW

File: Userland/Libraries/LibWeb/Geometry/DOMPoint.cpp [hunk 1]
```diff
@@ -20,6 +21,11 @@ DOMPoint::DOMPoint(JS::Realm& realm, double x, double y, double z, double w)
     set_prototype(&Bindings::cached_web_prototype(realm, "DOMPoint"));
 }
 
+JS::NonnullGCPtr<DOMPoint> DOMPoint::from_point(JS::VM& vm, Web::Geometry::DOMPointInit dom_point_init)
```
Comment (linusg) [inline Userland/Libraries/LibWeb/Geometry/DOMPoint.cpp:24]: https://drafts.fxtf.org/geometry/#dom-dompoint-frompoint

Comment (linusg) [reply Userland/Libraries/LibWeb/Geometry/DOMPoint.cpp:24]: The spec calls the argument `other`

File: Userland/Libraries/LibWeb/Geometry/DOMPoint.cpp [hunk 2]
```diff
@@ -20,6 +21,11 @@ DOMPoint::DOMPoint(JS::Realm& realm, double x, double y, double z, double w)
...
+{
+    return construct_impl(*vm.current_realm(), dom_point_init.x, dom_point_init.y, dom_point_init.z, dom_point_init.w);
```
Comment (linusg) [inline Userland/Libraries/LibWeb/Geometry/DOMPoint.cpp:26]: `// The fromPoint(other) static method on DOMPoint must create a DOMPoint from the dictionary other.`

File: Userland/Libraries/LibWeb/Geometry/DOMPointReadOnly.cpp [hunk 3]
```diff
@@ -24,6 +25,11 @@ DOMPointReadOnly::DOMPointReadOnly(JS::Realm& realm, double x, double y, double
     set_prototype(&Bindings::cached_web_prototype(realm, "DOMPointReadOnly"));
 }
 
+JS::NonnullGCPtr<DOMPointReadOnly> DOMPointReadOnly::from_point(JS::VM& vm, Web::Geometry::DOMPointInit dom_point_init)
```
Comment (linusg) [inline Userland/Libraries/LibWeb/Geometry/DOMPointReadOnly.cpp:28]: https://drafts.fxtf.org/geometry/#dom-dompointreadonly-frompoint

File: Userland/Libraries/LibWeb/Geometry/DOMPointReadOnly.cpp [hunk 4]
```diff
@@ -24,6 +25,11 @@ DOMPointReadOnly::DOMPointReadOnly(JS::Realm& realm, double x, double y, double
...
+{
+    return construct_impl(*vm.current_realm(), dom_point_init.x, dom_point_init.y, dom_point_init.z, dom_point_init.w);
```
Comment (linusg) [inline Userland/Libraries/LibWeb/Geometry/DOMPointReadOnly.cpp:30]: `// The fromPoint(other) static method on DOMPointReadOnly must create a DOMPointReadOnly from the dictionary other.`

File: Userland/Libraries/LibWeb/Geometry/DOMPointReadOnly.h [hunk 5]
```diff
@@ -12,13 +13,22 @@
...
+};
+
 // https://drafts.fxtf.org/geometry/#dompointreadonly
 class DOMPointReadOnly : public Bindings::PlatformObject {
     WEB_PLATFORM_OBJECT(DOMPointReadOnly, Bindings::PlatformObject);
 
 public:
     static JS::NonnullGCPtr<DOMPointReadOnly> construct_impl(JS::Realm&, double x = 0, double y = 0, double z = 0, double w = 1);
 
+    static JS::NonnullGCPtr<DOMPointReadOnly> from_point(JS::VM&, DOMPointInit);
```
Comment (linusg) [inline Userland/Libraries/LibWeb/Geometry/DOMPointReadOnly.h:30]: pass by const ref

**Review by linusg — CHANGES_REQUESTED** [review]

File: Userland/Libraries/LibIDL/IDLParser.cpp [hunk 6]
```diff
@@ -176,6 +176,10 @@ NonnullRefPtr<Type> Parser::parse_type()
         return adopt_ref(*new UnionType("", nullable, move(union_member_types)));
     }
 
+    bool unrestricted = lexer.consume_specific("unrestricted");
```
[the remark this answers is not in the document]
Comment (AtkinsSJ) [reply Userland/Libraries/LibIDL/IDLParser.cpp:179]: :man_facepalming: I need to pay more attention to merge conflicts. That was me adding it in the first place, too.

**Comment by AtkinsSJ** [comment]
- Added spec comments
- Renamed the field to `other`
- Passed it as `const&`
- Removed the commit with the duplicate `unrestricted`-parsing code

**Review by linusg — APPROVED** [review]

**Pull request merged**